### DISCUSSION TITLE
How do I plot Work Orders related to an Account?

### DISCUSSION THREAD

**Discussion by deejay-hub** [deejay-hub/timeline-lwc#247] [Q&A] [answered]
I am having an issue showing related Work Orders at the Account level when placing the Timeline. How do I map those to have them reflect?

**Comment by deejay-hub** [comment, marked as answer]
For further reading head to the [Configuration Guide](https://quip.com/ahFCA9VBEKtr) and [FAQs](https://quip.com/6yvoAcBukqZB). These docs help you understand how you can add custom objects, junction objects and other areas of customisation.

Time Warp uses a custom metadata setting to decide which child objects to plot.

Step 1: Head to Setup -> Custom Metadata Types
<img width="1904" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/fb0fc111-e617-4d73-b048-8063df4b6bc6">

Step 2: Click on the "Manage Records" action on the Custom Metadata Type called "Timeline Configuration"
<img width="1898" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/1430e2fc-5faa-43bb-ab33-a15e19e5f0bc">

Step 3: Make sure the default view is updated to show "All Configurations"
<img width="1911" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/d49b7852-9daa-4c47-92fa-7d06be5716cb">

Step 4: Locate the Label for Account_WorkOrders and click Edit.
<img width="1909" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/1c6187fa-9f8b-4a50-a38d-1b1ee6f6c870">

Step 5: Change the Active flag from false to true then click Save
<img width="1906" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/a9989db4-d098-48d8-a930-a729afe4a4c8">

Note that this will plot the Work Orders based on the date in the StartDate field of the Work Order record using the standard realtionship between Accounts and Work Orders. **Make sure the records have a value for StartDate on the records you want to plot**.

You can change the date used to plot the record by updating the "Position Date Field" in Step 5. To show a different text that appears next to the Icon update the "Detail Field". To change the fields in the hover then update the compact layout.

If you're using Person Accounts then you will have to also activate the following settings. This will show WorkOrders for a Person Account if they are in the AccountId of ContactId fields on the Work Order.
<img width="515" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/1edcdbc6-2c01-4d4f-9ee7-615397a5868b">

**Reply by KJohansen80** [reply]
Glorious tool you have created here, sir. Thank you very very much for the help. It is SO hard to find help with Salesforce without having to shell out thousands of dollars. 
One last question I now have the Work Orders plotting on the timeline, how do I get to see the associated Work Type in words and not the SF code? The Work Type is a related lookup field. If that is even possible...
Thank you again! Great work here my man!

![image](https://github.com/deejay-hub/timeline-lwc/assets/138698291/9b28839c-bb45-4dd1-a715-4daf46e0a8d6)
![image](https://github.com/deejay-hub/timeline-lwc/assets/138698291/c0c1c343-2724-4075-a08a-b8787dc50d99)
![image](https://github.com/deejay-hub/timeline-lwc/assets/138698291/e9b5e8a4-2379-4ee8-b2a1-4263d777c69f)

**Reply by KJohansen80** [reply]
UPDATE: I created a related lookup field on the Work Order that displays the worktype.name in the field. 
So I use tat for the Timeline subject and I am off to the races. Cheers my friend! Love the tool!

**Reply by deejay-hub** [reply]
Glad you got it working. Creating a field will work also....just for completeness you would also be able to specify the lookup in the Detail Field as follows. That saves you creating a new field.
<img width="728" alt="image" src="https://github.com/deejay-hub/timeline-lwc/assets/7021883/19df0862-db69-43f7-a907-e01c363ea56b">
